### PR TITLE
Add support for memory connector Range() tokens

### PR DESCRIPTION
**Pull request by rkuris**
This implements continuation tokens for the Range() operator in the
memory connector.  Just before returning from Range(), if the length
of the return value is longer than requested, we encode the data
in the last value returned to the user.  The encoding is a base64
encoded gob.

When a token is supplied on a future call to Range(), we still find
all the rows, and then decode the token, and look for that value
in the result set.  If we find it, then we just start at the next
value. If it's not there, that's okay too; we just start at the
next value in the set. The latter can only happen when the specific
value in the token has been deleted.

### CODE REVIEW

**Comment by rkuris** [comment]
Reviewers: this has been broken into three separate commits for ease of review, a cleanup commit 678ac94, the Range() work 6f9a2cf, and the Scan() work.

File: connectors/memory/memory_test.go [hunk 1]
```diff
@@ -707,3 +784,93 @@ func TestPanics(t *testing.T) {
...
+	sut := NewConnector()
+
+	// we don't use the token if there's no data that matches, so lets
+	// create one row
+	createTestData(t, sut, func(id int) string {
+		return "data"
+	}, 1)
+
+	token := "this is not a token and not a hot dog"
+	t.Run("testInvalidTokenRange", func(t *testing.T) {
```
Comment (rkuris) [inline connectors/memory/memory_test.go]: The other test, for range on scan, is nearly identical and shares the same setup. The error report when you fail a test is really nice too.

File: connectors/memory/memory.go [hunk 2]
```diff
@@ -534,15 +534,9 @@ func (c *Connector) Scan(_ context.Context, ei *dosa.EntityInfo, minimumFields [
...
-	if token != "" {
-		var err error
-		start, err = decodeToken(token)
-		if err != nil {
-			return nil, "", errors.Wrapf(err, "Invalid token %q", token)
-		}
-		startPartKey = partitionKeyBuilder(ei, start)
+	startPartKey, start, err := getStartingPoint(ei, token)
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "Invalid token %s", token)
```
Comment (klnusbaum) [inline connectors/memory/memory.go]: Nit, I think this wrap error message is incorrect. The `getStartingPoint` function could've failed for any number of reasons. I would make this wrap error message something like `failed to scan` and then (if you want) wrap the error from `decodeToken` down in `getStartingPoint`. In _that_ wrap, I would mention about the invalid token.

Comment (rkuris) [inline connectors/memory/memory.go]: Yes, that's true. I will fix it then merge.

**Pull request merged**